### DISCUSSION TITLE
emqx_docker_mongo_error

### DISCUSSION THREAD

**Discussion by uchauhan1994** [emqx/emqx#7197] [General]
I am using an emqx/emqx:4.4.1 docker image, I have done my custom configuration at emqx_auth_mongo.conf file, and the configuration file is also attached below.

Whenever I start the MongoDB plugin from the dashboard at that time, nothing happens, but when I log in to the broker application, on the console thousands of errors come.

Let me know If anything I missed

[emqx_auth_mongo.conf.txt](https://github.com/emqx/emqx/files/8159832/emqx_auth_mongo.conf.txt)
[error.txt](https://github.com/emqx/emqx/files/8159833/error.txt)
.

**Comment by uchauhan1994** [comment]
I am using the wrong way to connect MongoDB.
If you have a MongoDB cluster in your environment, then the below settings help you to connect with MongoDB.

##############################################################
emqx_auth_mongo.conf
##############################################################
auth.mongo.type = rs
auth.mongo.srv_record = false
auth.mongo.rs_set_name = rs0
auth.mongo.server = 127.0.0.1:27017,127.0.0.1:27017,127.0.0.1:27017
auth.mongo.pool = 8
auth.mongo.username = abcd
auth.mongo.password = abcd@123
auth.mongo.auth_source = mqtt
auth.mongo.database = mqtt
auth.mongo.query_timeout = 5s
auth.mongo.topology.pool_size = 1
auth.mongo.topology.max_overflow = 0
auth.mongo.auth_query.password_hash = sha256
auth.mongo.auth_query.collection = mqtt_user
auth.mongo.auth_query.password_field = password
auth.mongo.auth_query.selector = username=%u
auth.mongo.super_query.collection = mqtt_user
auth.mongo.super_query.super_field = is_superuser
auth.mongo.super_query.selector = username=%u
auth.mongo.acl_query.collection = mqtt_acl
auth.mongo.acl_query.selector = username=%u
##############################################################

**Comment by mononym** [comment]
So the issue here is that it looks like your mongodb server/cluster is set up with SCRAM authentication. Per that authentication scheme there is an authentication database which holds the users and the database needs to be specified on connection. See: 
https://docs.mongodb.com/manual/reference/system-users-collection/#mongodb-data-admin.system.users.db
https://docs.mongodb.com/manual/core/authentication/

The error coming back appears to be about not knowing which database to connect to for the user/not finding the specified users. Off the top of my head I don't know 100% how we handle this with configuration and passing the information through so I need to look into it and get back to you but it likely won't be until tomorrow.

I **think** that the correct field you need to use to configure the auth database is `auth_source` which I see you have configured. I'd look into what the actual database is supposed to be for you to specify for the authentication database and update that config value with that database and try again.

If that doesn't work the quick fix would be to remove auth from your mongodb cluster while you're still doing development and, as I said, I'll get to looking into this tomorrow to see if I can get you a final answer.

**Comment by mononym** [comment]
You might also try simply removing the `auth_source` declaration as it looks like the client will set the value to `admin` if it is not defined: https://github.com/emqx/mongodb-erlang/blob/master/src/connection/mc_worker.erl#L193

Edit: Nevermind, was confusing your setup with the default config you posted thinking the one you posted was actually yours rather than the file you attached being yours. So ignore this.

**Comment by mononym** [comment]
@[uchauhan1994](https://github.com/uchauhan1994) What version of mongo are you using?

**Reply by uchauhan1994** [reply]
I don't about it, but the issue has been fixed, and I have also written a solution to it in the comment.
Thanks for your response.
